config/compression.go

```diff
@@ -0,0 +1,32 @@
+package config
+
+import "strings"
+
+type Compression struct {
+	Request  CompressionInfo `mapstructure:"request"`
+	Response CompressionInfo `mapstructure:"response"`
+}
+
+// CompressionInfo defines what types of compressions are supported
+type CompressionInfo struct {
+	GZIP bool `mapstructure:"enable_gzip"`
+}
+
+type CompressionType string
+
+const (
+	CompressionGZIP CompressionType = "gzip"
+)
+
+func (k CompressionType) ToLower() CompressionType {
+	return CompressionType(strings.ToLower(string(k)))
+}
+
+func (cfg *CompressionInfo) IsSupported(compressionType CompressionType) bool {
```

Comment:
This is called only once where the argument is a HTTP Content Encoding. That implies `CompressionType` is actually masquerading as `HttpContentEncoding`. I recommend to separate these concepts and not handle the content encoding type here. It's a coincidence gzip is both our enum value and the http encoding type. There's no guarantee these types will continue to align in the future.